doc/features.md

```diff
@@ -0,0 +1,2601 @@
+
+@gmemoize
+def some_evens(n):
+    yield from drop(n, evens())
+```
+
+The only differences are the name of the decorator and ``return`` vs. ``yield from``. The point of `fimemoize` is that in simple cases like this, it allows us to use a regular factory function that makes an iterable, instead of a gfunc. Of course, the gfunc could have several `yield` expressions before it finishes, whereas the factory function terminates at the `return`.
+
+
+### ``fup``: Functional update; ``ShadowedSequence``
+
+We provide ``ShadowedSequence``, which is a bit like ``collections.ChainMap``, but for sequences, and only two levels (but it's a sequence; instances can be chained). Use ``ShadowedSequence`` for slicing (read-only), equality comparison, ``str`` and ``repr``. Out-of-range read access to a single item emits a meaningful error, like in ``list``. See its docstring for details.
+
+The function ``fupdate`` functionally updates sequences and mappings. Whereas ``ShadowedSequence`` reads directly from the original sequences at access time, ``fupdate`` makes a shallow copy (of the same type as the given input sequence) when it finalizes its output. The utility function ``fup`` is a specialization of ``fupdate`` to sequences, and adds support for the standard slicing syntax.
+
+First, let's look at ``fupdate``:
+
+```python
+from unpythonic import fupdate
+
+lst = [1, 2, 3]
+out = fupdate(lst, 1, 42)
+assert lst == [1, 2, 3]  # the original remains untouched
+assert out == [1, 42, 3]
+
+lst = [1, 2, 3]
+out = fupdate(lst, -1, 42)  # negative indices also supported
+assert lst == [1, 2, 3]
+assert out == [1, 2, 42]
+```
+
+Immutable input sequences are allowed. Replacing a slice of a tuple by a sequence:
+
+```python
+from itertools import repeat
+lst = (1, 2, 3, 4, 5)
+assert fupdate(lst, slice(0, None, 2), tuple(repeat(10, 3))) == (10, 2, 10, 4, 10)
+assert fupdate(lst, slice(1, None, 2), tuple(repeat(10, 2))) == (1, 10, 3, 10, 5)
+assert fupdate(lst, slice(None, None, 2), tuple(repeat(10, 3))) == (10, 2, 10, 4, 10)
+assert fupdate(lst, slice(None, None, -1), tuple(range(5))) == (4, 3, 2, 1, 0)
+```
+
+Slicing supports negative indices and steps, and default starts, stops and steps, as usual in Python. Just remember ``a[start:stop:step]`` actually means ``a[slice(start, stop, step)]`` (with ``None`` replacing omitted ``start``, ``stop`` and ``step``), and everything should follow. Multidimensional arrays are **not** supported.
+
+When ``fupdate`` constructs its output, the replacement occurs by walking *the input sequence* left-to-right, and pulling an item from the replacement sequence when the given replacement specification so requires. Hence the replacement sequence is not necessarily accessed left-to-right. (In the last example above, ``tuple(range(5))`` was read in the order ``(4, 3, 2, 1, 0)``.)
+
+The replacement sequence must have at least as many items as the slice requires (when applied to the original input). Any extra items in t
```

Comment:
Could delete these notes on old versions (0.10.0 and 0.11.1).

Reply:
Do you mean remove the paragraph?

Reply:
Yes, could remove the two paragraphs, the one starting with **v0.10.0** and the one starting with **v0.11.1**.